ui/apps/everest/.e2e/release/init-deploy.e2e.ts

```diff
@@ -0,0 +1,202 @@
+// everest
+// Copyright (C) 2023 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { expect, test } from '@playwright/test';
+import {
+  deleteDbCluster,
+  suspendDbCluster,
+  resumeDbCluster,
+  restartDbCluster,
+} from '../utils/db-clusters-list';
+import { getTokenFromLocalStorage } from '../utils/localStorage';
+import { getClusterDetailedInfo } from '../utils/storage-class';
+import {
+  moveForward,
+  submitWizard,
+  populateBasicInformation,
+  populateResources,
+  populateAdvancedConfig,
+  populateMonitoringModalForm,
+} from '../utils/db-wizard';
+import { EVEREST_CI_NAMESPACES } from '../constants';
+import { waitForStatus, waitForDelete } from '../utils/table';
+import { checkError } from '../utils/generic';
+import { deleteMonitoringInstance } from '../utils/monitoring-instance';
+
+const { MONITORING_URL, MONITORING_USER, MONITORING_PASSWORD } = process.env;
+let token: string;
+
+test.describe.configure({ retries: 0 });
+
+[
+  { db: 'psmdb', size: 1 },
+  { db: 'psmdb', size: 3 },
+  { db: 'pxc', size: 1 },
+  { db: 'pxc', size: 3 },
+  { db: 'postgresql', size: 1 },
+  { db: 'postgresql', size: 3 },
+].forEach(({ db, size }) => {
+  test.describe(
+    'Init deploy workflow',
+    {
+      // Create/stop/resume/delete cluster in different configurations
+      tag: '@release',
+    },
+    () => {
+      const clusterName = `${db}-${size}-deploy`;
+
+      let storageClasses = [];
+      let namespace = EVEREST_CI_NAMESPACES.EVEREST_UI;
+      let monitoringName = `${db}-${size}-pmm`;
+
+      test.beforeAll(async ({ request }) => {
+        token = await getTokenFromLocalStorage();
+
+        const { storageClassNames = [] } = await getClusterDetailedInfo(
+          token,
+          request
+        );
+        storageClasses = storageClassNames;
+      });
+
+      test(`Cluster creation with ${db} and size ${size}`, async ({
+        page,
+        request,
+      }) => {
+        test.setTimeout(size * 120000);
+
+        expect(storageClasses.length).toBeGreaterThan(0);
```

Comment:
Hi Tomislav!  :)
The script assumes that there will always be at least one storage class available. If no storage classes are present, the test will fail early. I might be missing something here, but to clarify, are we ensuring that at least one storage class is available for the end-to-end test to run? Is that right?

Thanks in advance for your clarification!

Reply:
Perhaps adding an error message could provide clearer feedback about the cause of the failure 🙂 

`if (storageClasses.length === 0) {
  throw new Error("No storage classes available. Please ensure the environment is properly set up.");
}
`

Reply:
We are not ensuring that at least one storage class is available, but usually all k8s environments have at least one. But if we accidentally find ourselves in such environment the check will fail and it should be self explanatory so I don't think we need to add something here.